tools/windows/install-help/cal/customactiondata.cpp

```diff
@@ -31,28 +34,19 @@ bool CustomActionData::init(const std::wstring &data)
         return false;
     }
 
-    // first, the string is KEY=VAL;KEY=VAL....
-    // first split into key/value pairs
-    std::wstringstream ss(data);
-    std::wstring token;
-    while (std::getline(ss, token, L';'))
-    {
-        // now 'token'  has the key=val; do the same thing for the key=value
-        bool boolval = false;
-        std::wstringstream instream(token);
-        std::wstring key, val;
-        if (std::getline(instream, key, L'='))
-        {
-            std::getline(instream, val);
-        }
+    WcaLog(LOGMSG_STANDARD, "Initializing CustomAction with %S\n", data.c_str());
 
-        if (val.length() > 0)
-        {
-            this->values[key] = val;
-        }
+    auto start = data.begin();
+    auto end = data.end();
+    std::wregex re(L"((\\w+)=(.+?);\r\n)");
```

Comment:
It might be a good idea to allow whitespace after the `;`, otherwise a stray space after a property in the XML will make this fail to match that line.